osu.Game/Rulesets/Judgements/Judgement.cs

```diff
@@ -95,6 +153,42 @@ protected virtual double HealthIncreaseFor(HitResult result)
         /// <returns>The numeric health increase of <paramref name="result"/>.</returns>
         public double HealthIncreaseFor(JudgementResult result) => HealthIncreaseFor(result.Type);
 
-        public override string ToString() => $"AffectsCombo:{AffectsCombo} MaxResult:{MaxResult} MaxScore:{MaxNumericResult}";
+        public override string ToString() => $"MaxResult:{MaxResult} MaxScore:{MaxNumericResult}";
+
+        public static double ToNumericResult(HitResult result)
```

Comment:
I feel slightly uneasy with this function. Expressing the scoring relative to greats might look elegant, but I think I'd still advocate for the scores to be whole numbers to not open ourselves up to floating-point issues. These fractions all have non-well-defined denominators when it comes to base-2 and *will* drift when summed cumulatively. The more objects (and the more distinct results), the bigger the drift.

For instance, summing `Judgement.ToNumericResult(HitResult.Meh)` 6 times yields not 1, but 0.99999999999999989.

Reply:
Yeah that’s a good point. Not super sure why I made it relative, maybe to match HP... Will revert.